packages/ember-application/lib/index.js

```diff
@@ -11,10 +12,19 @@ import {
   Resolver
 } from 'ember-application/system/resolver';
 import Application from 'ember-application/system/application';
+import Engine from 'ember-application/system/engine';
+import EngineInstance from 'ember-application/system/engine-instance';
 
 Ember.Application = Application;
 Ember.Resolver = Resolver;
 Ember.DefaultResolver = DefaultResolver;
 
+if (isEnabled('ember-application-engines')) {
+  Ember.Engine = Engine;
+
+  // Expose EngineInstance for easy overriding. Reanalyze whether to
+  // continue exposing it after feature flag is removed.
```

Comment:
I don't care which way we go, but we need to be consistent.  Either both are exposed, or neither are...